aws_doc_sdk_examples_tools/metadata.py

```diff
@@ -393,6 +406,24 @@ def parse(
     return examples, errors
 
 
+def validate_no_duplicate_api_examples(examples: List[Example], errors: MetadataErrors):
```

Comment:
This doesn't need to be List, just Iterable. (then no need to build a list when calling just to rip it apart again below)

Reply:
Interfaces over classes, yes! Python makes me lazy...